nomad/job_endpoint.go

```diff
@@ -1785,14 +1787,58 @@ func (j *Job) Plan(args *structs.JobPlanRequest, reply *structs.JobPlanResponse)
 	}
 
 	// Create the scheduler and run it
-	sched, err := scheduler.NewScheduler(eval.Type, j.logger, snap, planner)
+	schedEventsCh := make(chan interface{}, 1)
+	sched, err := scheduler.NewScheduler(eval.Type, j.logger, schedEventsCh, snap, planner)
```

Comment:
You can reuse the existing logging chan here and avoid a lot of complexity (eg the flag, coordinated shutdown, etc).

```suggestion
	sched, err := scheduler.NewScheduler(eval.Type, j.logger, j.srv.workersEventCh, snap, planner)
```